htdocs/core/lib/functions.lib.php

```diff
@@ -102,18 +102,21 @@ function getDoliDBInstance($type, $host, $user, $pass, $name, $port)
 /**
  * 	Get list of entity id to use.
  *
- * 	@param	string	$element	Current element
- *                              'societe', 'socpeople', 'actioncomm', 'agenda', 'resource',
- *                              'product', 'productprice', 'stock',
- *                              'propal', 'supplier_proposal', 'facture', 'facture_fourn',
- *                              'categorie', 'bank_account', 'bank_account', 'adherent', 'user',
- *                              'commande', 'commande_fournisseur', 'expedition', 'intervention', 'survey',
- *                              'contract', 'tax', 'expensereport', 'holiday', 'multicurrency', 'project',
- *                              'email_template', 'event',
- * 	@param	int	     $shared	1=Return id of current entity + shared entities (default), 0=Return id of current entity only
+ * 	@param	string	$element		Current element
+ *									'societe', 'socpeople', 'actioncomm', 'agenda', 'resource',
+ *									'product', 'productprice', 'stock',
+ *									'propal', 'supplier_proposal', 'facture', 'facture_fourn',
+ *									'categorie', 'bank_account', 'bank_account', 'adherent', 'user',
+ *									'commande', 'commande_fournisseur', 'expedition', 'intervention', 'survey',
+ *									'contract', 'tax', 'expensereport', 'holiday', 'multicurrency', 'project',
+ *									'email_template', 'event',
+ * 	@param	int		$shared			0=Return id of current entity only,
+ * 									1=Return id of current entity + shared entities (default),
+ * 									2=Return id of current entity OR master entity 1 value (eg. dictionnary share)
```

Comment:
We don't need the value 2.
Currently value 1 means current id + "other" entities and it is the module that manage multicompany that is responsible to decide what means "other". The module can decide that other is some entities x or y because user has activated sharing but can also decide that "other" means the master entity because the table is a dictionary table (table llx_c_*).
This has another interesting effect : You don't have to add, everywhere there is a dictionary use, a new parameter with content getEntity('dictionary_table', 2). This is useless. If the dictionary table read has 
the common test "where entity in getEntity($tablename, 1)" like everywhere, it is enough and it is more clear that responsability to decide which entity value is included or not is on external module. This open road to more different usage and multicompany module implementation. 

You can have same effect, without pollution of dolibarr code with the choice of which entity is took into account or not, with no need to add a new parameter, just add the test on getEntity(tablename, 1). This is the external module that will decide and return entity to take into account.